src/NServiceBus.AcceptanceTests/Satellites/When_ripping_out.cs

```diff
@@ -0,0 +1,174 @@
+        }
+
+        class Context : ScenarioContext
+        {
+            public bool MessageReceived { get; set; }
+            public bool OtherMessageReceived { get; set; }
+        }
+
+        class Endpoint : EndpointConfigurationBuilder
+        {
+            public Endpoint()
+            {
+                EndpointSetup<DefaultServer>(c =>
+                {
+                    var transport = c.ConfigureTransport();
+                    transport.Routing().RouteToEndpoint(typeof(MyOtherMessage), Conventions.EndpointNamingConvention(typeof(Receiver)));
+                    
+                    c.Pipeline.Replace("TransportReceiveToPhysicalMessageProcessingConnector", new BastardizingTheCoreForkConnector(OnMessage, OnError));
+                    
+                    c.DisableFeature<AutoSubscribe>();
+                    c.DisableFeature<Sagas>();
+                });
+            }
+
+            static Task<ErrorHandleResult> OnError(IIncomingPhysicalMessageContext arg)
+            {
+                return Task.FromResult(ErrorHandleResult.RetryRequired);
+            }
+
+            static async Task OnMessage(ITransportReceiveContext rawContext, IIncomingPhysicalMessageContext context)
+            {
+                var outgoingMessage = new OutgoingMessage(rawContext.Message.MessageId, rawContext.Message.Headers, rawContext.Message.Body);
+                var transportOperation = new TransportOperation(outgoingMessage, new UnicastAddressTag(Conventions.EndpointNamingConvention(typeof(Receiver))));
+                context.AddToBatch(transportOperation);
+
+                // just to show what crazy stuff we can do now
+                await context.Send(new MyOtherMessage()).ConfigureAwait(false);
+            }
+
+            class BastardizingTheCoreForkConnector : StageForkConnector<ITransportReceiveContext, IIncomingPhysicalMessageContext, IBatchDispatchContext>
+            {
+                Func<ITransportReceiveContext, IIncomingPhysicalMessageContext, Task> onMessage;
+                Func<IIncomingPhysicalMessageContext, Task<ErrorHandleResult>> onError;
+
+                public BastardizingTheCoreForkConnector(Func<ITransportReceiveContext, IIncomingPhysicalMessageContext, Task> onMessage, Func<IIncomingPhysicalMessageContext, Task<ErrorHandleResult>> onError)
+                {
+                    this.onError = onError;
+                    this.onMessage = onMessage;
+                }
+                
+                public override async Task Invoke(ITransportReceiveContext context, Func<IIncomingPhysicalMessageContext, Task> stage, Func<IBatchDispatchContext, Task> fork)
+                {
+                    var physicalMessageContext = this.CreateIncomingPhysicalMessageContext(context.Message, context);
+                    
+                    try
+                    {
+                        var pendingTransportOperations = new PendingTransportOperations();
```

Comment:
do we need this? Could we also connect this to the `IDispatchContext` stage directly and essentially do immediate dispatches (that what we would do with all the raw endpoints or satellites too?)

Reply:
Sure that is also possible. I went for the more luxurious option to demonstrate what is possible.

Reply:
good to know 👍